### PR TITLE
Implemented getsockname syscall

### PR DESCRIPTION
**Pull request by alfink**

### CODE REVIEW

File: qiling/os/posix/syscall/socket.py [hunk 1]
```diff
@@ -155,6 +155,24 @@ def ql_syscall_bind(ql, bind_fd, bind_addr, bind_addrlen,  *args, **kw):
     ql.os.definesyscall_return(regreturn)
 
 
+def ql_syscall_getsockname(ql, sockfd, addr, addrlen, *args, **kw):
+    if sockfd < 256 and ql.os.file_des[sockfd] != 0:
+        host, port = ql.os.file_des[sockfd].getsockname()
+        data = struct.pack("<h", 1)
+        data += struct.pack(">H", port)
+        data += ipaddress.ip_address(host).packed
+        data = data[:addrlen]
```
Comment (ucgJhe) [inline qiling/os/posix/syscall/socket.py:164]: addrlen is a pointer not scalar, you need to read it first before doing this

File: qiling/os/posix/syscall/socket.py [hunk 2]
```diff
@@ -155,6 +155,24 @@ def ql_syscall_bind(ql, bind_fd, bind_addr, bind_addrlen,  *args, **kw):
...
+        ql.mem.write(addr, data)
+        regreturn = 0
+    else:
+        regreturn = -1
+
+    ql.nprint("getsockname(%d, %x, %d) = %d" % (sockfd, addr, addrlen, regreturn))
```
Comment (ucgJhe) [inline qiling/os/posix/syscall/socket.py:170]: use 0x%x for pointers will be better for keeping consistency in qiling

File: qiling/os/posix/syscall/socket.py [hunk 3]
```diff
@@ -155,6 +155,24 @@ def ql_syscall_bind(ql, bind_fd, bind_addr, bind_addrlen,  *args, **kw):
     ql.os.definesyscall_return(regreturn)
 
 
+def ql_syscall_getsockname(ql, sockfd, addr, addrlen, *args, **kw):
+    if sockfd < 256 and ql.os.file_des[sockfd] != 0:
+        host, port = ql.os.file_des[sockfd].getsockname()
+        data = struct.pack("<h", 1)
```
Comment (ucgJhe) [inline qiling/os/posix/syscall/socket.py:161]: is there any reason we keep this field constant value ?

**Comment by xwings** [comment]
@ucgJhe thanks for the help :)

**Comment by alfink** [comment]
@ucgJhe Thanks for the feedback. All you mentioned should be fixed now

**Comment by xwings** [comment]
please make sure all PR goes into dev not master

**Comment by xwings** [comment]
thanks for the PR

**Pull request merged**